qiskit/pulse/transforms.py

```diff
@@ -518,14 +517,24 @@ def align_func(schedule: Schedule,
     return pad(aligned, aligned.channels, until=duration, inplace=True)
 
 
-def flatten(program: ScheduleComponent) -> ScheduleComponent:
-    """Flatten (inline) any called nodes into a Schedule tree with no nested children."""
-    if isinstance(program, instructions.Instruction):
+def flatten(program: PulseProgram) -> PulseProgram:
+    """Flatten (inline) any called nodes into a Schedule tree with no nested children.
+
+    Args:
+        program: Pulse program to remove nested structure.
+
+    Returns:
+        Flatten pulse program.
+
+    Raises:
+        PulseError: When invalid data format is given.
+    """
+    if isinstance(program, ScheduleBlock):
         return program
```

Comment:
Yes, For example,

```python
with build() as sched:
    with align_left():
        play(Gaussian(...), DriveChannel(1))
        with align_right():
            play(Gaussian(...), DriveChannel(0))
            play(Gaussian(...), DriveChannel(1))
```

This is an example of nested block. As you can see the right align block cannot be flattened because parent context has left transform.